services/proposals/src/propose.py

```diff
@@ -36,40 +99,29 @@ def propose(db_insert_fn: Callable[[Mapping[T, T], T], None]) -> None:
     try:
         with db.raw_pdfs.watch([{'$match': {'operationType': 'insert'}}]) as stream:
             for doc in stream:
-                full = doc['fullDocument']
-                obj_id = str(full['_id'])
-                doc_num = int(obj_id, 16)
-                if doc_num % replica_count != pod_id:
-                    continue
-                logging.info('Document found and added to queue')
-                if 'page_data' not in full:
-                    full['page_data'] = []
-                page_data = full['page_data']
-                for page in db.pages.find({'pdf_id' : full['_id']}):
-                    tpage = {}
-                    for i in ['page_width', 'page_height', 'page_num']:
-                        tpage[i] = page[i]
-                    bstring = page['resize_bytes']
-                    img = Image.open(io.BytesIO(bstring)).convert('RGB')
-                    coords = get_proposals(img)
-                    tpage['proposals'] = coords
-                    page_data.append(tpage)
-                db_insert_fn(full, client)
+                propose_doc(doc, db, db_insert_fn, multimachine=True, replica_count=replica_count, pod_id=pod_id)
     except pymongo.errors.PyMongoError as err:
         logging.error("Error in pymongo:")
         logging.error(err)
 
     end_time = time.time()
     logging.info(f'Exiting proposal generation. Time up: {end_time - start_time}')
 
-def mongo_insert_fn(obj: Mapping[T, T], client: T) -> None:
+def mongo_insert_fn(objs, client):
     db = client.pdfs
-    cc_pdfs = db.cc_pdfs
-    result = cc_pdfs.insert_one(obj)
+    pages = db.propose_pages
+    #result = pages.replace_one({'_id':ObjectId(record['_id'])}, obj, upsert=True)
+    result = pages.insert_many(objs)
     logging.info(f"Inserted result: {result}")
 
 
+@click.command()
+@click.argument('num_processes')
```

Comment:
Again, not a huge deal, but this is incompatible with the helm deployment method.